packages/strapi-admin/admin/src/containers/Roles/ListPage/index.js

```diff
@@ -1,18 +1,113 @@
-import React from 'react';
-import { Link } from 'react-router-dom';
-import { useGlobalContext } from 'strapi-helper-plugin';
+import React, { useState, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import { Button } from '@buffetjs/core';
+import { List, Header } from '@buffetjs/custom';
+import { Plus } from '@buffetjs/icons';
+import { useGlobalContext, ListButton } from 'strapi-helper-plugin';
 
-const ListPage = () => {
-  const { settingsBaseURL } = useGlobalContext();
+import RoleRow from './RoleRow';
+import ListWrapper from './ListWrapper';
+import BaselineAlignment from './BaselineAlignment';
+
+const RoleListPage = () => {
+  const { settingsBaseURL, formatMessage } = useGlobalContext();
+  const { push } = useHistory();
+  const [selectedRoleIds, setSelectedRoleIds] = useState([]);
+  const [roles] = useState([
```

Comment:
Why don't you use a reducer here? It will be easier to add tests and also to synchronise your data.

Reply:
Yes, I will handle all the data with a reducer. This is temporary until we have the back end API.